manifests/osfamily/debian.pp

```diff
@@ -66,7 +66,7 @@
           ensure   => absent,
           priority => '90',
         }
-      } elsif $puppet_agent::collection =~ /core/ {
+      } elsif $puppet_agent::collection and $puppet_agent::collection =~ /core/ {
```

Comment:
why do you check if $puppet_agent::collection isn't undef? params.pp sets collection always, and the datatype in init.pp enforces `String`, so this shouldn't be required and the old syntax was fine?

Reply:
Edit: in the commit message you say:

> If `$puppet_agent::collection` was undef, then the regex match would result in an error

But was is there a situation where `undef` is possible?

Reply:
Another question: Why a regex? Are there valid values like `core8`, `core-latest`? Could you maybe document them in the init.pp?

Reply:
> so this shouldn't be required and the old syntax was fine?

I removed the commit.

> Why a regex? Are there valid values like core8, core-latest? Could you maybe document them in the init.pp?

They're documented in the task metadata:

```ruby
Optional[Enum[puppet7, puppet8, puppet, puppet7-nightly, puppet8-nightly, puppet-nightly, puppetcore7, puppetcore8]]
```

But in puppet's `init`, it is defined to just be a String. I'll update the documentation for that parameter.